src/scripts/getEmissionsFromEmpreinte.ts

```diff
@@ -85,35 +86,57 @@ const validStatus = ['Valide générique', 'Valide spécifique', 'Archivé']
 
 const escapeTranslation = (value?: string) => (value ? value.replaceAll('"""', '') : value)
 
-const saveEmissions = async (url: string) => {
+const getUnit = (value?: string): Unit | null => {
+  if (!value) {
+    return null
+  }
+  if (value.startsWith('kgCO2e/')) value = value.replace('kgCO2e/', '')
+  if (value.toLowerCase() === 'tep pci') value = 'tep PCI'
+  if (value.toLowerCase() === 'tep pcs') value = 'tep PCS'
+  if (value.toLowerCase() === 'ha') value = 'ha'
+  if (value.toLowerCase() === 'kg') value = 'kg'
+  if (value.toLowerCase() === 'litre') value = 'litre'
+  if (['kWh (PCI)', 'KWh PCI', 'kWhPCI'].includes(value)) value = 'kWh PCI'
+  if (value === 'kWhPCS') value = 'kWh PCS'
+  if (value === 'm2 SHON') value = 'm² SHON'
+  if (value.includes('m3')) value = value.replace('m3', 'm³')
+
+  const unit = Object.entries(UNITS_MATRIX).find((entry) => entry[1] === value)
+  if (unit) return unit[0] as Unit
+  throw new Error('Unit not found for ' + value)
+}
+
+const saveEmissions = async (url: string, posts: any[]) => {
```

Comment:
pq any ?

Reply:
Parce qu'en définissant posts: `EmissionResponse['results'][]`, on obtient une erreur au moment du posts.push(emission) : `Argument of type '{ "Identifiant_de_l'\u00E9l\u00E9ment": string; "Statut_de_l'\u00E9l\u00E9ment": string; Type_Ligne: string; Source: string; Type_poste: string; Localisation_géographique: string; 'Sous-localisation_g\u00E9ographique_fran\u00E7ais': string; ... 27 more ...; Qualité_C: number; }' is not assignable to parameter of type '{ "Identifiant_de_l'\u00E9l\u00E9ment": string; "Statut_de_l'\u00E9l\u00E9ment": string; Type_Ligne: string; Source: string; Type_poste: string; Localisation_géographique: string; 'Sous-localisation_g\u00E9ographique_fran\u00E7ais': string; ... 27 more ...; Qualité_C: number; }[]'.
  Type '{ "Identifiant_de_l'\u00E9l\u00E9ment": string; "Statut_de_l'\u00E9l\u00E9ment": string; Type_Ligne: string; Source: string; Type_poste: string; Localisation_géographique: string; 'Sous-localisation_g\u00E9ographique_fran\u00E7ais': string; ... 27 more ...; Qualité_C: number; }' is missing the following properties from type '{ "Identifiant_de_l'\u00E9l\u00E9ment": string; "Statut_de_l'\u00E9l\u00E9ment": string; Type_Ligne: string; Source: string; Type_poste: string; Localisation_géographique: string; 'Sous-localisation_g\u00E9ographique_fran\u00E7ais': string; ... 27 more ...; Qualité_C: number; }[]': length, pop, push, concat, and 35 more.ts(2345)`

Reply:
EmissionResponse['results'] ?

Reply:
Je peux définir un type EmissionResponseElement et type EmissionResponse devient  
```
type EmissionResponse = {
  total: number
  next?: string
  results: EmissionResponseElement[]
}
```

Reply:
je comprends pas, EmissionResponse['results']  marche tres bien